onmt/Loss.py

```diff
@@ -103,6 +104,106 @@ def compute_loss(self, batch, output, target, **kwargs):
 
         return loss, stats
 
+    # TODO should the weights be used to affect "num_correct"?
+    def stats(self, loss, scores, target):
+        """
+        Compute and return a Statistics object.
+
+        Args:
+            loss(Tensor): the loss computed by the loss criterion.
+            scores(Tensor): a sequence of predict output with scores.
+        """
+        pred = scores.max(1)[1]
+        non_padding = target.ne(self.padding_idx)
+        num_correct = pred.eq(target) \
+                          .masked_select(non_padding) \
+                          .sum()
+        return onmt.Statistics(loss[0], non_padding.sum(), num_correct)
```

Comment:
If the  `NMTLossComputeDatumWeighted` needs a new `stats()` method,  just override the base class version in `NMTLossComputeDatumWeighted`,  I think we don't need to put it in  the `NMTLossCompute` itself.

Reply:
Besides,  to be consistent with existing `*LossCompute` classes,  it would be better call it `DatumWeightedNMTLossCompute`

Reply:
I'm not sure about the `stats()` really... it was more of an open question and placeholder. I think the stats are what they should be, but maybe someone had an oppinion about weighting the `num_correct` or something. What's your take on that?

Reply:
For the `NMTLossCompute`, if we use the current `ONMTDataset`, the `stats()` in the base class would do,  no need to re-implement it in `NMTLossCompute`; if we use the weighted `ONMTDataset`, I think we should use`DatumWeightedNMTLossCompute`, not the `NMTLossCompute`, right?(If someone specifies -train_dw and -valid_dw, but still use the standard `NMTLossCompute`, it is weird, and I think we should add constraint here to not let user do so).  Again, no need to touch  `NMTLossCompute`.  So I am wondering why you put it in `NMTLossCompute`.

Then comes to question whether we need a `weighted` version of `stats()` for  the `DatumWeightedNMTLossCompute`?  We can think a little bit. The purpose of incorporating a `weighted` version of source data is due to the different degree of trustworthiness of each data example, I would interpret this as sort of data augmentation. But for the `num_correct` in `stats()`, it just tells how many words we predict correctly compared to the baseline `target`. In this sense, 1 is just 1,  I think we don't need to `weight` this figure here.

Reply:
you are right... there is no need so far to affect `stats()`, and it was in the wrong class anyway! just removed it on the latest commit.